boba-chain-ops/genesis/check.go

```diff
@@ -174,6 +191,59 @@ func PostCheckMigratedDB(
 	return nil
 }
 
+// PostCheckBobaProxyContracts will append the old slots to the ExpectedStorageSlots and ignore the
+// legacy proxy slots. This is because the legacy slots are affected by the migration process.
+func PostCheckBobaProxyContracts(g *types.Genesis) error {
+	for addr := range BobaUntouchablePredeploys {
+		for key, val := range g.Alloc[addr].Storage {
+			if key == ether.BobaLegacyProxyOwnerSlot || key == ether.BobaLegacyProxyImplementationSlot {
+				continue
+			}
+			ExpectedStorageSlots[addr][key] = val
+		}
+	}
+	return nil
+}
+
+// PostCheckBobaLegacyProxyImplementation will check that the legacy proxy implementation
+// contracts are replaced by the proxy contract and storage are wiped out except the admin slot.
+func PostCheckBobaLegacyProxyImplementation(tx kv.Tx) error {
```

Comment:
can you help me understand this -
I see the legacy implementation contracts are deleted (0x4200000000000000000000000000000000000021), with storage except the admin slot. why do we delete this? the implementation contracts shouldn't have storage slots set (including an admin slot)? do we want to redeploy new implementations? what are the legacy implementation contracts replaced by?

Reply:
Yes, the new implementation contracts are redeployed to a 0xcd... address. The legacy implementation contracts are replaced by proxy contract with the admin slot. https://github.com/bobanetwork/v3-anchorage/blob/boba-develop/packages/contracts-bedrock/contracts/universal/Proxy.sol
We can set the implementation contract to our own contract to use this proxy address.

Reply:
Optimism generates 2048 proxy contracts for the future usage. 
https://github.com/bobanetwork/v3-anchorage/blob/16e8aeb0c2e5fb11ca7aabc592f2f5cdc40e77df/op-chain-ops/genesis/setters.go#L78

We can still use `0x4200000000000000000000000000000000000021` for the implementation contract if we want to.

Reply:
ok got it, so the new implementation will be on the 0xcd.. address